### PR TITLE
Hrm04 module linking UI fixes

### PR DESCRIPTION
**Pull request by akshaya86**
- module linking
- ui fixes
- digital watch face service added

### CODE REVIEW

**Comment by EntyR** [comment]
Please add snap helper to center central element

**Comment by EntyR** [comment]
Also please compare ui design with one from task document. It must look similar

**Comment by EntyR** [comment]
Design template doesn't categories it's properly so I'll put it here

1st template its normal state and the rest 2 it's pressed state.

Also please watch for spacing between elements and for their size to fit design.

Spacing is a little bit big and elements are small

image

**Comment by EntyR** [comment]
Please lighten up the central element. Can change transparency of non-central for example.

Check for elements color and font it should fit those in design template 

Also look for elements size. Only 3 element should be on the screen at the time. 

image

**Comment by akshaya86** [comment]
@EntyR Please revisit this PR

**Comment by EntyR** [comment]
It's not building now. Check IGetHeartRateUseCase  interface

**Comment by EntyR** [comment]
image
Please update spacing to match figma template, also check for pressed and not central elements design it's differ

**Comment by EntyR** [comment]
image
Please also look for top and bottom elements:

1. It shouldn't clip any of them like in screenshot
2. you should be able to center 1'st element

**Comment by EntyR** [comment]
image
Text is clipping

**Comment by EntyR** [comment]
Look if you can use [different dimen resources](https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes) or calculate size with screen height and width for devices with bigger screen size

image

**Comment by EntyR** [comment]
Please move watch face and others screen to appropriate requests if you not going to implement them here, 
also tell me if you are going to do this here, I'll test them too then

**Comment by akshaya86** [comment]
Hi @EntyR i will cover all UI fixes and changes in this PR only. 
Working on latest review comments.

**Comment by akshaya86** [comment]
@EntyR please revisit this PR.

I have tested watch screen of WearOS Large Round 30API with resoultion 454*454

**Comment by EntyR** [comment]
image
It seems like where no data loaded for graph so can't test this one, still please fix positioning for element to fit figma as I descripted in the issue.

About sizing you always should use either  different [dimen resource](https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes) files for other screen size or calculate size programmatically in activity/fragment instead based on your device screen size, basically your design and design from figma should fit as much as possible and different screen sizes kind of ruin this idea, it's not an issue in this project but still nice thing to have

**Comment by EntyR** [comment]
As for main screen please add margin for top element so it can be centered properly

**Comment by akshaya86** [comment]
@EntyR Please revisit this PR

**Comment by EntyR** [comment]
image
Still unable to center first and last element it should look like this. Try to update margin in adapter onbind viewholder method something like this 


```
val layoutParams = holder.binding.root.layoutParams as ViewGroup.MarginLayoutParams
if (currentFeature.id == 0) {
            layoutParams.updateMargins(top =myoffset)
        }
```

**Comment by EntyR** [comment]
Testing on 320x320 resolution device

image

items too small 4 elements could fit on the screen and where should be only 3

**Comment by EntyR** [comment]
Also please reduce time of waiting for export data 1 second is enough we just need to show what the screen exist and not force users to actually wait

**Comment by EntyR** [comment]
Very nice Measure screen, but still toast messages not looking too great 

image

Please replace with something like dialog box or just change text in start measure textview same with toast for other screen.

**Comment by akshaya86** [comment]
@EntyR  Thank you for revisiting . i will retest screen with 320*320 , update the PR

**Comment by akshaya86** [comment]
@EntyR 

- Fixed all review comment also tested on emulator having resolution 384*384 as 382 is not available .
- Fixed margin issue and kept each item in centered position .
- reduced delay time 
- Removed all toast and added text error messages for respective screens .

File: app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt [hunk 1]
```diff
@@ -4,41 +4,25 @@ import android.content.BroadcastReceiver
...
-class MyWatchFaceDigital : CanvasWatchFaceService() {
+import java.text.SimpleDateFormat
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+class MyDigitalWatchFace : CanvasWatchFaceService() {
 
     companion object {
         private val NORMAL_TYPEFACE = Typeface.create(Typeface.SANS_SERIF, Typeface.NORMAL)
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt]: Please use roboco font like in template

File: app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt [hunk 2]
```diff
@@ -175,40 +160,53 @@ class MyWatchFaceDigital : CanvasWatchFaceService() {
...
-                    Toast.makeText(applicationContext, R.string.message, Toast.LENGTH_SHORT)
-                        .show()
+                    startActivity(Intent(this@MyDigitalWatchFace,MainActivity::class.java))
+                }
+
             }
             invalidate()
         }
 
         override fun onDraw(canvas: Canvas, bounds: Rect) {
```
Comment (EntyR) [inline app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt:173]: Please don't create any instances inside on draw or if you do make sure it's created only once, onDraw is called very often so it's bad practice to anything except drawing

**Comment by EntyR** [comment]
Also for your watchface background you can use gradient instead of loading image resources, something like this

`private val colors = intArrayOf( Color.BLACK, Color.BLACK, context.getColor(R.color.purple_primary), Color.BLACK, )`
`val backgroundColor = SweepGradient( bounds.exactCenterX(), bounds.exactCenterY(), colors, null )`

`val matrix = Matrix()`
`matrix.preRotate(27f, bounds.exactCenterX(), bounds.exactCenterY()) backgroundColor.setLocalMatrix(matrix) mainPaint.shader = backgroundColor`

File: app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt (hunk 1 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt]: Resolved

File: app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt (hunk 2 above)
Comment (akshaya86) [reply app/src/main/java/com/example/wearableapp/presentation/watchface/MyDigitalWatchFace.kt:173]: Not getting comment. can you please beat explain ? regarding background & time set ?

**Review by AlexeyKolosov — APPROVED** [review]

**Pull request merged**